node_modules/col-core/lib/api.js

```diff
@@ -51,9 +52,61 @@ var init = module.exports.init = function(callback) {
  */
 var initializeServer = function(callback) {
   // Initialize the Express server
-  module.exports.appServer = Server.setUpServer();
-  // Initialize the REST router wrapper
-  module.exports.appRouter = Server.setupRouter(module.exports.appServer);
+  var appServer = module.exports.appServer = Server.setUpServer();
+
+  // A router for all routes on /lti
+  var ltiRouter = module.exports.ltiRouter = express.Router();
+  appServer.use('/lti', ltiRouter);
+
+  // A router for all routes on /api/courses/:api_domain_course_id
+  var apiRouter = module.exports.apiRouter = express.Router();
+
+  // This middleware will check if the user is authorized to interact with the given course
+  apiRouter.use(function(req, res, next) {
+    // The base URL in this middleware will be /api/courses/ucberkeley.canvas.com~21312
+    // Unfortunately, express doesn't parse the api_domain and course in middleware, so
+    // we need to do some manual string mangling
+    var apiDomainCourseId = req.baseUrl.substring(13);
+    var apiDomain = apiDomainCourseId.split('~')[0];
+    var courseId = apiDomainCourseId.split('~')[1];
+
+    // Get the user id from the client's cookies
+    var cookieName = apiDomain + '_' + courseId;
+    var userId = req.signedCookies[cookieName];
+
+    // If no user id could be found, we bail out immediately
+    if (!userId) {
+      return res.status(401).send('Incorrect cookie information present');
+    }
+
+    // Get the user from the database so it can be made available on the request
+    var options = {
+      'where': {
+        'id': userId
+      },
+      'include': [DB.Course]
+    };
+    DB.User.findOne(options).complete(function(err, user) {
+      if (err) {
+        return res.status(500).send('Failed to look up a user');
+      } else if (!user) {
+        return res.status(401).send('Failed to find a user');
+      }
+
+      // Create a context on the request that contains some information
+      // about the user, domain and course
+      req.ctx = {
+        'user': user,
+        'apiDomain': apiDomain,
+        'course_id': courseId,
```

Comment:
Do we need to add both the course id and the course? The course would contain the course_id as well, right ?